### ISSUE TITLE
How dependent destroy on comments will work

### ISSUE THREAD

**Issue by tarungarg** [lml/commontator#111]
I want to delete all comments thread as my post delete. Please tell me how can we dependent destroy the comments

**Comment by mulander** [comment]
There appears to be no support so far for a way to automatically destroy the thread + comments when the commontable object gets destroyed.

There is however a simple way to implement such behaviour. Marking something as `acts_as_commontable` adds the following `has_one` relation

```ruby
          has_one :thread, as: :commontable,
                           class_name: 'Commontator::Thread'
```

so @tarungarg let's assume you have a `Swing` model that people can comment on, and you want the comment thread and all existing comments being deleted when the swing gets removed. To do so you can re-define the above relation with `dependent: :destroy` in your model after including `acts_as_commontable` as in the example below:

```ruby
class Swing < ApplicationRecord

  acts_as_commontable

  has_one :thread, as: :commontable,
    class_name: 'Commontator::Thread',
    dependent: :destroy
end
```

internally `Commontator::Thread` already defines it's comments and subscriptions with `dependent: :destroy` so no further changes are required.

@Dantemss this looks like a rather straightforward thing to implement as an additional parameters to `acts_as_commontable`. The `options` hash is already taken for `Commontator::CommontableConfig` but I think this is a common enough use-case to warrant another parameter changing the signature to `acts_as_commontable(options = {}, dependent = nil)`. If that's desirable I would be willing to implement this (I am actually using the provided above solution in my project).

**Comment by Dantemss** [comment]
I think I would prefer if you just made `dependent` a special case in the options hash, so something like:

```
association_options = options.slice(:dependent)
config_options = options.except(:dependent)
```

**Comment by mulander** [comment]
I updated the PR with your suggestion.

**Comment by Dantemss** [comment]
Released version 5.1.0 with this feature. Thanks @mulander.
